front/src/modules/simulationResult/components/SpeedSpaceChart/helpers.ts

```diff
@@ -0,0 +1,155 @@
+    },
+    value:
+      identifier && sncf ? `${identifier.name} ${sncf.ch !== ('' || '00') ? sncf.ch : ''}` : '',
+  }));
+
+export const formatElectrifications = (
+  electrifications: PathPropertiesFormatted['electrifications']
+) =>
+  electrifications.map(({ electrificationUsage, start, stop }) => ({
+    position: {
+      start: mToKm(start),
+      end: mToKm(stop),
+    },
+    value: {
+      type: electrificationUsage.type,
+      voltage:
+        'voltage' in electrificationUsage
+          ? (electrificationUsage.voltage as ElectrificationValues['voltage'])
+          : undefined,
+      lowerPantograph:
+        'lower_pantograph' in electrificationUsage ? electrificationUsage.lower_pantograph : false,
+    },
+  }));
+
+export const formatSlopes = (slopes: PositionData<'gradient'>[]) => {
+  let actualSlopes: LayerData<number>[] = [];
+  if (slopes)
+    actualSlopes = slopes.map((slope, index) => ({
+      position: {
+        start: mToKm(index === 0 ? 0 : slopes[index - 1].position),
+        end: mToKm(slope.position),
+      },
+      value: slope.gradient,
+    }));
+  return actualSlopes;
+};
+
+export const getProfileValue = (
+  electricalProfileType: 'profile' | 'no_profile',
+  profile: keyof typeof electricalProfilesDesignValues | null,
+  voltage?: '1500V' | '25000V',
+  handled?: boolean
+) => {
+  if (electricalProfileType === 'no_profile') {
+    return { electricalProfile: 'neutral' };
+  }
+  if (!handled || profile === null) {
+    return {
+      electricalProfile: 'incompatible',
+      color:
+        voltage === '1500V'
+          ? electricalProfilesDesignValues.F.color
+          : electricalProfilesDesignValues['20000V'].color,
+    };
+  }
+  return {
+    electricalProfile: profile,
+    ...electricalProfilesDesignValues[profile!],
+  };
+};
+
+export const formatElectricalProfiles = (
+  simulation: SimulationResponseSuccess,
+  electrifications: Data['electrifications']
+) => {
+  if (simulation.electrical_profiles.values.length === 0) return undefined;
+  return simulation.electrical_profiles.values.map((values, index) => {
+    const electrification = electrifications.find(
+      (electrificationValue) =>
+        electrificationValue.position.start >=
+          (index === 0 ? 0 : mmToKm(simulation.electrical_profiles.boundaries[index - 1])) &&
+        electrificationValue.position.end! <=
+          mmToKm(simulation.electrical_profiles.boundaries[index])
+    );
+
+    return {
+      position: {
+        start: mmToKm(index === 0 ? 0 : simulation.electrical_profiles.boundaries[index - 1]),
+        end: mmToKm(simulation.electrical_profiles.boundaries[index]),
+      },
+      value: getProfileValue(
+        values.electrical_profile_type,
+        'profile' in values
+          ? (values.profile as keyof typeof electricalProfilesDesignValues)
+          : null,
+        electrification?.value.voltage,
+        'handled' in values ? values.handled : true
+      ),
```

Comment:
If you did the previous comment refacto

```suggestion
   value: getProfileValuev2(values, electrification?.value.voltage),
```

Reply:
done